packages/pyright-internal/src/analyzer/docStringUtils.ts

```diff
@@ -55,53 +55,133 @@ export function cleanAndSplitDocString(rawString: string): string[] {
-        if (paramOffset >= 0) {
-            return trimmedLine.substr(paramOffset + 7);
+    const docstringTypes = [
+        { marker: `@param ${paramName}`, trim: 7 },
+        { marker: `:param ${paramName}:`, trim: 7 },
+        { marker: `${paramName}:`, trim: 0 },
+        { marker: `${paramName} (`, trim: 0 },
+    ];
+
+    // Iterate over each line and check if it matches one of the above markers.
+    for (const [idx, line] of docStringLines.entries()) {
+        let ret = undefined;
+        let paramOffset = -1;
+
+        for (const typ of docstringTypes) {
+            // If we get a match, process the remaining lines of the docstring to check for multi-line parameter descriptions.
+            paramOffset = line.indexOf(typ.marker);
+            if (paramOffset >= 0) {
+                ret = parseParameterDescription(paramOffset, typ.trim, docStringLines.slice(idx));
+                if (ret) {
+                    return ret;
+                }
+            }
         }
+    }
 
-        // Check for Google (variant 1) format
-        paramOffset = trimmedLine.indexOf(paramName + ': ');
-        if (paramOffset >= 0) {
-            return trimmedLine.substr(paramOffset);
-        }
+    return undefined;
+}
 
-        // Check for Google (variant 2) format
-        paramOffset = trimmedLine.indexOf(paramName + ' (');
-        if (paramOffset >= 0) {
-            return trimmedLine.substr(paramOffset);
+// Iterate over the remainder of the docstring lines and look for multi-line parameter descriptions.
+//
+// By the time we're in this function, we know that the first entry in the given slice
+// contains a match for the parameter of interest. We process that, and then check if subsequent
+// lines are a continuation of the description on the first line. If so, we concatenate them
+// and return the full parameter description
+function parseParameterDescription(paramOffset: number, trim: number, docStringLines: string[]): string | undefined {
+    let ret = undefined;
+    for (const line of docStringLines) {
+        // we know the first entry is the first line of the param description
+        if (ret === undefined) {
+            ret = line.slice(paramOffset + trim).trimEnd();
+        } else {
+            const initialSpaces = line.match(/(^ +)\S/);
+            if (initialSpaces) {
+                // description continuation should have more leading spaces than the
+                // first line of the description
+                if (initialSpaces[1].length > paramOffset) {
+                    ret += ' ' + line.trim();
+                } else {
+                    return ret;
+                }
+            } else if (line.length !== 0) {
+                // if the line doesn't start with spaces and isn't empty, it's not a continuation of this param's description
+                return ret;
+            } else {
+                // empty line, keep looking (TODO: should we just stop here?)
```

Comment:
hmm, yeah i think either we should just stop here, or we should update it to show the newline when hovering over the parameter name.

i don't really mind either way, so whichever is easiest

Reply:
Yeah I've been using this for a little while now and I still feel like I could go either way. If you have a really long description for a parameter (which, arguably, you shouldn't), it's kinda nice to be able to insert a real line break for clarity. You could still achieve this by just inserting a `\n` though so maybe that's sufficient? Idk, it's kinda nice that you could just insert a real line break and keep the paragraph going as long as you maintain the indentation...

Reply:
yeah i don't mind which solution we go with but i do find it annoying when i'm writing some markdown using newlines and it gets rendered all on one line unexpectedly, so i think we should probably decide one one of these approaches before merging

Reply:
Ah gotcha, misunderstood your original comment. Yeah I think keeping the newline in would make sense, I can make that change.